### PR TITLE
fixes pango_markup support with title_format

### PR DESCRIPTION
**Pull request by baloo**
The support for pango_markup was broken in title_format because the
formated title was escaped. I think only the payload should be escaped.

This commit fixes 789a877b379cd35c350610be62b971ae00feb542

The pango_markup looks like:
```
font pango:"DejaVu Sans Mono 10"
for_window [class=".*"] title_format "<span background='#ff0000'>%title</span>"
```

### CODE REVIEW

File: sway/tree/view.c [hunk 1]
```diff
@@ -832,6 +832,25 @@ static size_t append_prop(char *buffer, const char *value) {
...
+	escape_markup_text(buffer, escaped_title);
+	free(buffer);
+	return escaped_title;
+}
+
+static const char *escape_if_pango(const char *buffer) {
+	// if using pango_markup in font, we need to escape all markup char
+	// from values to avoid messing with pango markup
+	if (!config->pango_markup) {
+		return escape_title(strdup(buffer));
```
Comment (emersion) [inline sway/tree/view.c]: This leaks the value returned by `escape_title`. Maybe we should move this to `append_prop`?

Comment (baloo) [reply sway/tree/view.c]: fixed as suggested

File: sway/tree/view.c [hunk 2]
```diff
@@ -824,12 +824,31 @@ struct sway_view *view_from_wlr_surface(struct wlr_surface *wlr_surface) {
 	return NULL;
 }
 
+static char *escape_title(char *buffer) {
```
Comment (emersion) [inline sway/tree/view.c]: We could make `buffer` a `const char*` and stop `free`-ing it in this function.

File: sway/tree/view.c [hunk 3]
```diff
@@ -824,12 +824,31 @@ struct sway_view *view_from_wlr_surface(struct wlr_surface *wlr_surface) {
...
 static size_t append_prop(char *buffer, const char *value) {
 	if (!value) {
 		return 0;
 	}
-	lenient_strcat(buffer, value);
-	return strlen(value);
+	// if using pango_markup in font, we need to escape all markup char
+	// from values to avoid messing with pango markup
+	if (!config->pango_markup) {
+		char * escaped_value = escape_title(strdup(value));
```
Comment (emersion) [inline sway/tree/view.c]: Style: no space between `*` and variable name.

**Comment by emersion** [comment]
Thanks!

**Pull request merged**